### ISSUE TITLE
Docs for the segmentation training format

### ISSUE THREAD

**Issue by PonteIneptique** [mittagessen/kraken#139]
Hey there,
I'd like to train a segmenter before leaving next week, and I understood you will release it soon. Any place where I can find the segmenter training format ?

**Comment by mittagessen** [comment]
The segmenter is trained on baselines, so the training data are input images and the baselines (in practice any continuous line drawn through the text line works) contained therein. The input images are just any image (b/w, grayscale, rgb all work although I've mostly used rgb), the target are JSON files with the extension `.path` of nested lists with each list representing the points of each (base-)polyline:

```
[
[[x0, y0], [x1, y1], [x2, y2], [x3, y3], ...], # line 0
[[x25, y25], [x26, y26]], # line 1
....
]
```

The training data for the Arabic model is [here](http://homer.dh.uni-leipzig.de/badam.tar) (4.4Gb, freely distributable). There's still a small bug in the kraken training code which I hope I'll have fixed by the end of the day.

**Comment by PonteIneptique** [comment]
Thanks, we already had data from Transkribus. With 240 * 46 lines training (and ~24 eval), we only achieve 0.64 which feels low, any input on that ?

**Comment by PonteIneptique** [comment]
(as a small note, just in case, scikit-image is not part of the requirements in `blla` branch although it's imported)

**Comment by mittagessen** [comment]
The extraction isn't completely implemented, yet as I have to rewrite quite a bit of the existing segmenter/recognizer code to make it work. I can do a quick hack to make the VGSL models work with the development code in https://github.com/mittagessen/blla if that helps.

For the metrics: They just measure pixelwise accuracy which is not perfect when you want to know how good the actual detected baselines are except that higher is usually better. Have a look at [this](https://ieeexplore.ieee.org/iel7/8394761/8395092/08395221.pdf) article for a better (although still flawed) method of determining the quality of a baseline detection. There's a java implementation around but you're probably better advised just to look at the pixels yourself to get a picture.

**Comment by PonteIneptique** [comment]
Yes, it could definitely help as we want to have both Kraken and Transkribus data for our article at DH :)

When I'll be able to visualize the result, I'll be able to tell you how well it worked (note that this is manuscript, maybe you trained first for print)

**Comment by mittagessen** [comment]
If you only need it for an article I'd say just use the code in the separate blla repository, even for training. That one is somewhat guaranteed to work while the code in kraken is mostly untested right now.

I trained on both the cBad dataset (Latin manuscripts) and beat all methods apart from [ARU-net](https://github.com/TobiasGruening/ARU-Net) (although I'm mostly sure the difference lies in the postprocessing and not the classifier itself), and on the (highly complex and varied) dataset of ~40 different Arabic-script manuscripts linked above with an F1 of ~0.92.

**Comment by PonteIneptique** [comment]
I assume you mean this ? https://github.com/mittagessen/blla/blob/rec/blla/main.py

**Comment by mittagessen** [comment]
Yup. That's what I'm porting over into the kraken codebase right now.

BTW depending on the nature of your manuscripts it might be worthwile to pretrain on the [cBad dataset](https://zenodo.org/record/3234502)

**Comment by PonteIneptique** [comment]
So, I'd train and retrain on my own dataset ?

**Comment by PonteIneptique** [comment]
You would not happen to have such a pretrain model around ? Would save me some time ;)

**Comment by mittagessen** [comment]
There's one pretrained on the simple track at [link](http://homer.dh.uni-leipzig.de/cbad_recnet.pth).

**Comment by mittagessen** [comment]
As a note, these have been pretrained with a scaled line width of 10 pixels which might reduce the recall a bit as multiple lines are merged. The default 4 pixels in ketos are a bit low so if you use kraken it's best to set it to something between 5-9.

**Comment by mrocr** [comment]
@mittagessen Can we talk numbers here, What is the baseline detector:
- model number of parameters.
- prediction fps.
- neural architecture: resnet-50, espnet , enet... ?
- prediction ram usage.
- training time.

**Comment by mrocr** [comment]
@PonteIneptique Can you close this topic.

**Comment by rohanchauhan01** [comment]
> EDIT: I had good experiences with [labelme](https://github.com/wkentaro/labelme) as an annotation tool but others have failed miserably at installing it even with lots of handholding, so your mileage might vary.
Hello @mittagessen, 

I am trying to work with [labelme](https://github.com/wkentaro/labelme), and have annotated a few complex layout scans using it. But it only allows to save data in .json. Is there a quick way to convert this to the template in docs/pagexml.xml?
 
or
Can I feed the labelme .json annotation to ``` ketos train ``` to get the desired result? 

and 
Is there a way to custom model using this annotation metadata in labelme's .json output? 

Best!

**Comment by dstoekl** [comment]
why not use eScriptorium <https://gitlab.inria.fr/scripta/escriptorium/-/issues> for annotation?

**Comment by rohanchauhan01** [comment]
Thanks for your suggestion. I haven't tried it. Now, I think, I will.
